sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -490,10 +621,26 @@ trait DivModLike extends BinaryArithmetic {
       s"${eval2.value} == 0"
     }
     val javaType = CodeGenerator.javaType(dataType)
-    val operation = if (operandsDataType.isInstanceOf[DecimalType]) {
-      decimalToDataTypeCodeGen(s"${eval1.value}.$decimalMethod(${eval2.value})")
+    val checkOverflow = if (operandsDataType.isInstanceOf[DecimalType]) {
```

Comment:
this name is confusing. This includes the operation and overflow checking. Maybe the previous `operation` is still a good name here.

Reply:
changed back